include/aspect/material_model/damage_rheology.h

```diff
@@ -0,0 +1,495 @@
+                                      const Point<dim> &position) const;
+
+        virtual double thermal_expansion_coefficient (const double      temperature,
+                                                      const double      pressure,
+                                                      const std::vector<double> &compositional_fields,
+                                                      const Point<dim> &position) const;
+        /**
+         * Rate of grain size growth (Ostwald ripening) or reduction
+         * (due to phase transformations) in dependence on temperature
+         * pressure, strain rate, mineral phase and creep regime.
+         * We use the grain size evolution laws described in Solomatov
+         * and Reese, 2008. Grain size variations in the Earth’s mantle
+         * and the evolution of primordial chemical heterogeneities,
+         * J. Geophys. Res., 113, B07408.
+         */
+        virtual
+        double
+        grain_size_growth_rate (const double                  temperature,
+                                const double                  pressure,
+                                const std::vector<double>    &compositional_fields,
+                                const SymmetricTensor<2,dim> &strain_rate,
+                                const Tensor<1,dim>          &velocity,
+                                const Point<dim>             &position,
+                                const unsigned int            phase_index,
+                                const int                     crossed_transition) const;
+
+        /**
+         * Function that defines the phase transition interface
+         * (0 above, 1 below the phase transition).This is done
+         * individually for each transition and summed up in the end.
+         */
+        virtual
+        double
+        phase_function (const Point<dim> &position,
+                        const double temperature,
+                        const double pressure,
+                        const unsigned int phase) const;
+
+        /**
+         * Function that returns the phase for a given
+         * position, temperature, pressure and compositional
+         * field index.
+         */
+        virtual
+        unsigned int
+        get_phase_index (const Point<dim> &position,
+                         const double temperature,
+                         const double pressure) const;
+
+        /**
+         * Function that takes an object in the same format
+         * as in.composition as argument and converts the
+         * vector that corresponds to the grain size to its
+         * logarithms and back and limits the grain size to
+         * a global minimum.
+         * @in normal_to_log: if true, convert from the grain
+         * size to its logarithm, otherwise from log to grain
+         * size
+         */
+        virtual
+        void
+        convert_log_grain_size (const bool normal_to_log,
```

Comment:
This function is never called with the argument `normal_to_log = true`, as far as I can see. We just need it to find out what the real grain size is, based on the compositional field. So you can just remove the argument, and this case from the implementation.